### PR TITLE
document helm chart

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Deployed version [here](https://garyverhaegen-da.github.io/daml-docs/2021-11-10-ef2e25af1a); mostly these two pages:

- [Connect Helm Chart](https://garyverhaegen-da.github.io/daml-docs/2021-11-10-ef2e25af1a/ops/connect/helm.html)
- [Setting Up Auth0](https://garyverhaegen-da.github.io/daml-docs/2021-11-10-ef2e25af1a/ops/connect/auth0.html)

**Comment by realvictorprm** [comment]
Stop working so late xD!

Gary Verhaegen ***@***.***> schrieb am Mi., 10. Nov. 2021,
23:44:

File: docs/source/ops/connect/helm.rst [hunk 1]
```diff
@@ -0,0 +1,281 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _connect-helm-chart:
+
+Connect Helm Chart
+==================
+
+As of 1.18.0, we provide an Early Access version of the Connect Helm Chart for
+Entreprise Edition customers. This page contains documentation for that Helm
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:10]:
```suggestion
Enterprise Edition customers. This page contains documentation for that Helm
```
france

File: docs/source/ops/connect/helm.rst [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+chart.
+
+Credentials
+-----------
+
+Like all Entreprise Edition components, the Helm Chart is hosted on
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:16]:
```suggestion
Like all Enterprise Edition components, the Helm Chart is hosted on
```
baguette

File: docs/source/ops/connect/helm.rst [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+Artifactory. To get both the Helm chart itself and the Docker images it relies
+on, you will need Artifactory credentials. In the rest of thie document, we
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:18]:
```suggestion
on, you will need Artifactory credentials. In the rest of the document, we
```

File: docs/source/ops/connect/helm.rst [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+----------------
+
+There are many options you may want to set for a production setup. See the
+reference at the end of this page for full details. At a minimum, though, you
+need to set the following:
+
+- ``production=true``: By default, the Helm chart starts a number of components
+  that are meant to give you a quick idea of what the Helm chart enables, but
+  are most definitely not meant for production use. Specifically, this will
+  disabble the internal PostgreSQL instances, the mock auth server, and the
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:153]:
```suggestion
  disable the internal PostgreSQL instances, the mock auth server, and the
```

File: docs/source/ops/connect/helm.rst [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+`Elasticsearch <https://www.elastic.co/elasticsearch/>`,
+`Logstash <https://www.elastic.co/logstash/>`_,
+`Kafka <https://kafka.apache.org/>`_, etc.
+
+You can find external documentation on, how to setup `ElasticSearch` with
+`Filebeat` and `Kibana` for analyzing logs on your Kubernetes cluster
+`here <https://www.deepnetwork.com/blog/2020/01/27/ELK-stack-filebeat-k8s-deployment.html>`_.
+
+As of 1.18.0, the :ref:`HTTP JSON API <json-api>` component in the Helm chart produces
+JSON-encoded logs. Other components log in somewhat less structured formats.
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:196]:
```suggestion
JSON-encoded logs. Other components log as unstructured text.
```

File: docs/source/ops/connect/helm.rst [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+Daml Metrics Options
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:198]: The capitalization of title is inconsistent. I would standardize on capitalizing only given names (like "Helm") and nothing else (like "metrics" and "options") -- apart from the first word of course.

File: docs/source/ops/connect/helm.rst [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+--------------------
+
+The Daml Driver for PostgreSQL instance and the HTTP JSON API instances started
+by the Helm chart are configured to expose Prometheus metrics on a port named
+`metrics`, using the appropriate annotations. This means that, if you are
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:203]:
```suggestion
``metrics``, using the appropriate annotations. This means that, if you are
```

File: docs/source/ops/connect/helm.rst [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+If you are running the Helm chart solely for the HTTP JSON API Service
+(connected to an external Ledger API server), then you can eschew backing up
+entirely, as the database for the HTTP JSON API Service is an
+easy-to-reconstruct cache. This assume that, in this setup, the data store of
+the Ledger API server is, itself, properly backed up.
+
+Securing Daml Connect
+---------------------
+
+The Helm chart assumes that the Kubernetes environment itself is trusted, and
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:258]: Speaking about capitalization, check whether "chart" should be capitalized. To me a "chart" is some visual representation of data and "Chart" is the given name of Helm bundles (I'm quite curious of why they chose that name though).

File: docs/source/ops/connect/helm.rst [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+does, in which case all of the communications between the TLS termination
+endpoint must be happening on a fully trusted network.
+
+See the :ref:`auth0` section for an example of setting up nginx to proxy
+external connections to the JSON API, Trigger Service and OAuth2 Middleware.
+
+Helm Chart Options Reference
+----------------------------
+
+TODO: copy over Configuration.md
```
Comment (stefanobaghino-da) [inline docs/source/ops/connect/helm.rst:281]: In this PR?

**Review by stefanobaghino-da — APPROVED** [review]
Looks like I learned how to spell "enterprise" in French.

File: docs/source/json-api/production-setup.rst [hunk 10]
```diff
@@ -316,4 +316,4 @@ A meter. Number of package uploads
 ``daml.http_json_api.allocation_party_throughput``
 --------------------------------------------------
 
-A meter. Number of party allocations
\ No newline at end of file
+A meter. Number of party allocations
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply docs/source/json-api/production-setup.rst:319]: Newline at end of file, I think.

File: docs/source/ops/connect/auth0.rst [hunk 11]
```diff
@@ -649,19 +582,189 @@ Next, create a file ``nginx/Dockerfile`` with this content:
...
+****************************
+
+For simplicity, we assume that you have access to a server with a public IP
+address that both you and Auth0 can reach. Furthermore, we assume that you have
+access to Entreprise Edition credentials to download the Docker images.  We
+also assume you can create a local cluster with ``minikube`` on the remote
+machine. Finally, we assume that you have downloaded the Helm chart in a folder
+called ``daml-connect``.
+
+First, start a new cluster::
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply docs/source/ops/connect/auth0.rst:611]: Yes, that makes the next thing fixed-size if it's indented.

File: docs/source/ops/connect/helm.rst [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+
+where ``values.yaml`` is a YAML file that includes at least the
+``imagePullSecret`` key. See the rest of this page for other options in
+``values.yaml``, and the Helm documentation for related Helm usage.
+
+Setting up the ``imagePullSecret``
+----------------------------------
+
+The Helm chart relies on the production-ready Docker images for individual
+components that are part of the Entreprise Edition. Specifically, it expects a
```
Comment (realvictorprm) [inline docs/source/ops/connect/helm.rst:60]: Just do a search n replace for `Entreprise` to `Enterprise` :joy:

File: docs/source/ops/connect/auth0.rst [hunk 13]
```diff
@@ -649,19 +582,189 @@ Next, create a file ``nginx/Dockerfile`` with this content:
...
+
+Next, load up your credentials as explained in the :ref:`connect-helm-chart`
+section. We assume they are loaded under the secret named
+``daml-docker-credentials``.
+
+Create a file called ``values.yaml`` with the following content:
+
+.. code-block:: yaml
+
+   imagePullSecret: daml-docker-credentials
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply docs/source/ops/connect/auth0.rst:623]: Ultimately it's a text file that people can change however they want, but if they do use the provided script to set up the credentials, this is the name they'll end up with.

File: docs/source/ops/connect/helm.rst [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+- :ref:`JSON API metrics <json-api-metrics>`
+
+Upgrading
+---------
+
+.. note::
+
+   This section only makes sense with the ``production`` flag set to ``true``.
+
+Upgrading the Daml Connect version should be done by uninstalling the exiting
```
Comment (realvictorprm) [inline docs/source/ops/connect/helm.rst:219]: `existing` instead of `exiting`

File: docs/source/ops/connect/helm.rst [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+version. Destroying all of the components is a safe operation because all of
+the state is stored in the provided database coordinates. There is no
+additional state within the components themselves.
+
+The components are not designed for running concurrently with older versions,
+so it is imperative to wait for the existing Helm chart components to be
+completely shut down before installing the new one. Do not try to upgrade in
+place.
+
+Assuming you do not change the database coordinates, you should have data
```
Comment (realvictorprm) [inline docs/source/ops/connect/helm.rst:230]: only concerned about the json api here but in theory it should be fine

File: docs/source/ops/connect/helm.rst (hunk 9 above)
Comment (realvictorprm) [reply docs/source/ops/connect/helm.rst:281]: Wondering too

**Review by realvictorprm — APPROVED** [review]
OK looks very good to me

File: docs/source/ops/connect/helm.rst (hunk 6 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:198]: I was going for title case, which I believe is the proper way to write titles in English. I did miss two of them (the last ones I added; I might have already been half asleep at that point), but Ithink calling the whole thing inconsistent is a bit harsh. 😿

File: docs/source/ops/connect/helm.rst (hunk 9 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:281]: Yes, but I was out of fuel for a md -> rst translation yesterday.

File: docs/source/ops/connect/helm.rst (hunk 8 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:258]: Looking through the helm.sh site, I can only find one instance of "Helm Chart" outside a title, and lots of instances of "Helm chart" through the text. I do think it looks a bit weird too, but :shrug:.

**Comment by garyverhaegen-da** [comment]
> Looks like I learned how to spell "enterprise" in French.

I always thought "entreprise" was the British version, like "centre" v. "center". Oh well. I'll update.

File: docs/source/ops/connect/auth0.rst [hunk 16]
```diff
@@ -649,19 +582,189 @@ Next, create a file ``nginx/Dockerfile`` with this content:
...
+  PROXY="$(minikube ip):$(kubectl get svc dm-daml-connect-reverse-proxy --output=json | jq '.spec.ports[0].nodePort')"
+  docker run -e JSON_IP=$PROXY \
+             -e AUTH_IP=$PROXY/auth \
+             -e TRIGGER_IP=$PROXY/trigger \
+             -e FRONTEND_IP=$DOMAIN \
+             --network=host \
+             frontend
+
+where ``$DOMAIN`` is assumed to be an environment variable set to the public
+domain on which your server is exposed. And voilà! Your application is up and
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply docs/source/ops/connect/auth0.rst:672]: Permission to add that to the actual docs page?

File: docs/source/ops/connect/helm.rst (hunk 5 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:196]: That was my first version, but it's not actually completely unstructured, is it? Anyway, done.

File: docs/source/ops/connect/auth0.rst [hunk 17]
```diff
@@ -649,19 +582,189 @@ Next, create a file ``nginx/Dockerfile`` with this content:
...
+We document two paths forward here: one that relies on the Helm chart included
+in Daml Connect Entreprise Edition, and a manual setup using only the Community
+Edition SDK.
+
+Using the Connect Helm Chart
+****************************
+
+For simplicity, we assume that you have access to a server with a public IP
+address that both you and Auth0 can reach. Furthermore, we assume that you have
+access to Entreprise Edition credentials to download the Docker images.  We
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply docs/source/ops/connect/auth0.rst]: Done.

File: docs/source/ops/connect/helm.rst (hunk 1 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Done.

File: docs/source/ops/connect/helm.rst (hunk 2 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Done.

File: docs/source/ops/connect/helm.rst (hunk 4 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Done.

File: docs/source/ops/connect/helm.rst (hunk 6 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:198]: I've updated the two miserably-failing titles to title case, so it's now all consistent. If you really don't like title case I guess we can go with something else.

File: docs/source/ops/connect/helm.rst (hunk 7 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Done, thanks.

File: docs/source/ops/connect/helm.rst (hunk 12 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Yes, that's what I did. 🙂

File: docs/source/ops/connect/helm.rst (hunk 14 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst]: Thanks!

File: docs/source/ops/connect/helm.rst (hunk 15 above)
Comment (garyverhaegen-da) [reply docs/source/ops/connect/helm.rst:230]: Would you mind expanding on that concern?

Comment (realvictorprm) [reply docs/source/ops/connect/helm.rst:230]: after reading it again I think it is clear enough

File: docs/source/ops/connect/helm.rst (hunk 6 above)
Comment (stefanobaghino-da) [reply docs/source/ops/connect/helm.rst:198]: > but Ithink calling the whole thing inconsistent is a bit harsh. :crying_cat_face:

Sorry, I didn't mean to be harsh. Thanks for making it all consistent. I don't think any choice is bad, as long as only one approach is chosen, thanks.

**Pull request merged**